tangelo/toolboxes/unitary_generator/unitary.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module defines the unitary abstract class, providing the foundation to
+implement unitary evolution circuits.
+"""
+
+import abc
+from typing import Union, List
+
+
+class Unitary(abc.ABC):
+    """Base class for all Unitary circuit definitions. Derived/children classes wirtten by users and
+    developers must implement the following abstract methods.
+    """
+
+    def __init__(self):
+        pass
+
+    @abc.abstractmethod
+    def qubit_indices(self):
+        """Return the indices used by the algorithm to propagate the unitary.
+
+        Returns:
+            List[int]: State qubits
+            List[int]: Ancilla qubits
+        """
+        pass
+
+    @abc.abstractmethod
+    def build_circuit(self, n_steps: int, control: int = Union[None, int, List[int]]):
+        """Build and return the quantum circuit implementing the unitary evolution for n_steps.
+
+        Args:
+            n_steps(int): The number of unitary evolution steps with control
```

Comment:
what about "control" ?

Reply:
I am not sure this comment was resolved: you still have 3 args in your signature but only 1 documented in the docstring